crates/steel-core/src/rvals.rs

```diff
@@ -2124,8 +2124,7 @@ impl fmt::Display for SteelVal {
         // at the top level, print a ' if we are
         // trying to print a symbol or list
         match self {
-            SymbolV(_) | ListV(_) => write!(f, "'")?,
-            VectorV(_) => write!(f, "'#")?,
```

Comment:
:question: This is a change in behavior:

```
;; before
λ > (write 'a)
'a

;; after
λ > (write 'a)
a
```

I think it kinda makes sense for `write` to output something that can be parsed back with `read`. AFAICS that is not a requirement of R7RS, but it is the [documented behavior](https://docs.racket-lang.org/reference/Writing.html#%28def._%28%28quote._~23~25kernel%29._write%29%29) in Racket:

> Writes datum to out, normally in such a way that instances of core datatypes can be read back in

Reply:
Hm, I might have broken some tests because of this :thinking:

Reply:
So I'll have to look at the tests; its possible that the tests need to be updated since they're either still using the old behavior or were just fundamentally wrong to begin with.

I think that matching behavior with Racket in this case is good - in fact, I'm surprised that I didn't check that it was matching before. In any case I'm okay with this behavior change, assuming we can make sure it doesn't break other things / update everything else to work with it

Reply:
Just needed to remove the `'` from a few expectations :+1: 

> AFAICS that is not a requirement of R7RS

I just noticed that it is a suggestion in R6RS.